### PR TITLE
Update googletest and benchmark libraries, remove gcov link dependency

### PR DESCRIPTION
**Pull request by k06a**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,8 @@ SET (boolinq_VERSION_MAJOR 2)
 SET (boolinq_VERSION_MINOR 0)
 
 
-SET (CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS} -O0 -ggdb3 -DDEBUG")
-SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3")
+SET (CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS} -O0 -ggdb3 -DDEBUG -std=c++11")
+SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3 -std=c++11")
```
Comment (babelvit) [inline CMakeLists.txt]: I can not pull this pull-request to proove it, but i gues the `-std=c++11` will not let me to build again. What about using `gnu++11`?

Comment (k06a) [inline CMakeLists.txt]: @babelvit added this arguments to fix updated google externals/benchmark/test/cxx03_test:
```
set(CXX03_FLAGS "${CMAKE_CXX_FLAGS}")
string(REPLACE "-std=c++11" "-std=c++03" CXX03_FLAGS "${CXX03_FLAGS}")
string(REPLACE "-std=c++0x" "-std=c++03" CXX03_FLAGS "${CXX03_FLAGS}")
```

So I can really use `-std=c++11` or `-std=c++0x`. What do you think?

Comment (k06a) [inline CMakeLists.txt]: Reverted this changes

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,8 +5,8 @@ SET (boolinq_VERSION_MAJOR 2)
 SET (boolinq_VERSION_MINOR 0)
 
 
-SET (CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS} -O0 -ggdb3 -DDEBUG -std=c++11")
-SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3 -std=c++11")
+SET (CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS} -O0 -ggdb3 -DDEBUG")
+SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O3")
```
Comment (babelvit) [inline CMakeLists.txt]: I deduce that your compiler does not have `-std=gnu++11` option, or you don't like it. After reconsideration, I leave it to your decision. It is not difficult to make the little correction locally at all.

Comment (babelvit) [inline CMakeLists.txt]: The current state builds here OK.

**Comment by k06a** [comment]
@babelvit ok, I'll merge this PR. Do you have experience in configuring cpp project for travis-ci.org?

**Comment by babelvit** [comment]
@k06a No, I don't any experience with it.

Sorry for the mystification, it does not build without some correction. But don't worry. When it is one or two corrections, I can cope with it.

**Pull request merged**